cmd/signozschemamigrator/schema_migrator/traces_migrations.go

```diff
@@ -1,3 +1,360 @@
+			},
+			CreateTableOperation{
+				Database: "signoz_traces",
+				Table:    "distributed_traces_v3_resource",
+				Columns: []Column{
+					{Name: "labels", Type: ColumnTypeString, Codec: "ZSTD(5)"},
+					{Name: "fingerprint", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "seen_at_ts_bucket_start", Type: ColumnTypeInt64, Codec: "Delta(8), ZSTD(1)"},
+				},
+				Engine: Distributed{
+					Database:    "signoz_traces",
+					Table:       "traces_v3_resource",
+					ShardingKey: "cityHash64(labels, fingerprint)",
+				},
+			},
+			CreateTableOperation{
+				Database: "signoz_traces",
+				Table:    "trace_summary",
+				Columns: []Column{
+					{Name: "traceID", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "first_reported",
+						Type: SimpleAggregateFunction{
+							FunctionName: "min",
+							Arguments:    []ColumnType{DateTime64ColumnType{Precision: 9}},
+						},
+						Codec: "ZSTD(1)"},
+					{Name: "last_reported",
+						Type: SimpleAggregateFunction{
+							FunctionName: "max",
+							Arguments:    []ColumnType{DateTime64ColumnType{Precision: 9}},
+						},
+						Codec: "ZSTD(1)"},
+					{Name: "num_spans",
+						Type: SimpleAggregateFunction{
+							FunctionName: "sum",
+							Arguments:    []ColumnType{ColumnTypeUInt64},
+						},
+						Codec: "ZSTD(1)"},
+				},
+				Engine: AggregatingMergeTree{
+					MergeTree: MergeTree{
+						PartitionBy: "toDate(first_reported)",
+						OrderBy:     "(traceID)",
+						TTL:         "toDateTime(first_reported) + toIntervalSecond(1296000)",
+						Settings: TableSettings{
+							{Name: "index_granularity", Value: "8192"},
+							{Name: "ttl_only_drop_parts", Value: "1"},
+						},
+					},
+				},
+			},
+			CreateTableOperation{
+				Database: "signoz_traces",
+				Table:    "distributed_trace_summary",
+				Columns: []Column{
+					{Name: "traceID", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "first_reported",
+						Type: SimpleAggregateFunction{
+							FunctionName: "min",
+							Arguments:    []ColumnType{DateTime64ColumnType{Precision: 9}},
+						},
+						Codec: "ZSTD(1)"},
+					{Name: "last_reported",
+						Type: SimpleAggregateFunction{
+							FunctionName: "max",
+							Arguments:    []ColumnType{DateTime64ColumnType{Precision: 9}},
+						},
+						Codec: "ZSTD(1)"},
+					{Name: "num_spans",
+						Type: SimpleAggregateFunction{
+							FunctionName: "sum",
+							Arguments:    []ColumnType{ColumnTypeUInt64},
+						},
+						Codec: "ZSTD(1)"},
+				},
+				Engine: Distributed{
+					Database:    "signoz_traces",
+					Table:       "trace_summary",
+					ShardingKey: "cityHash64(traceID)",
+				},
+			},
+			CreateMaterializedViewOperation{
+				Database:  "signoz_traces",
+				ViewName:  "trace_summary_mv",
+				DestTable: "trace_summary",
+				Query: `SELECT
+							traceID,
+							minSimpleState(timestamp) AS first_reported,
+							maxSimpleState(timestamp) AS last_reported,
+							sumSimpleState(toUInt64(1)) AS num_spans
```

Comment:
1. -SimpleState suffix is not needed.
2. It should be count() instead of `sumSimpleState(toUInt64(1))` otherwise, you are under calculating the number of spans.

Reply:
1. You are suggeting it because it won't make any difference right?
2. Ahh right, thanks for correcting this.

Reply:
>You are suggeting it because it won't make any difference right?

Yes